components/nimbus/Cargo.toml

```diff
@@ -35,6 +35,7 @@ once_cell = "1"
 uniffi = { version = "^0.18", optional = true }
 chrono = { version = "0.4", features = ["serde"]}
 unicode-segmentation = "1.8.0"
+glean = { path = "../external/glean/glean-core/rlb" }
```

Comment:
Hmmm I'll need to think about this a little, from an "impact on android" perspective.

Two questions:
1) is always guaranteed that if the glean binary isn't initialized, all glean API's are no-ops? Because that will be the case here in android, where they use libxul
2) How OK are we with duplicating the glean binary in android?

I'm not sure about the first question, and the second question we can maybe measure?

Alternatively, for now, we can gate including glean on a feature https://doc.rust-lang.org/cargo/reference/features.html

we can even have it enabled by default, then in the [megazord for android](https://github.com/mozilla/application-services/blob/main/megazords/full/Cargo.toml) we can turn the feature off

Reply:
The answer to the first question is yes, uninitialized Glean is a no-op except for the queue for metrics recorded prior to init, but once it is full it too becomes a no-op. I don't expect much overhead from Glean in this regard and I am confident in Glean's logic for this.

Adding the RLB is just adding the binding layer on top of glean-core, so it's a fairly thin shim. There should still only be one glean-core coming from libxul and the rlb will interact with it on Android, and interact with the megazorded glean-core on iOS.  I didn't notice any big size regressions while testing in Focus, but happy to do a more detailed measurement if product has any issues with it.

Reply:
Ideally, I still would like to see either a [`cfg` on android](https://doc.rust-lang.org/reference/conditional-compilation.html#target_os) or some type of cargo feature that protects non-ios consumers from including a duplicate glean (and helps if anyone is confused why calls are no-ops) - if that does seem like a lot of work for what we would remove when glean-in-rust hits android I'm willing to concede, I'll leave it as a nice to have as long as android still builds and works OK